### PR TITLE
[cli/paraformer]   ali-paraformer load and infer work

### PR DESCRIPTION
**Pull request by Mddct**
Paraformer can infer in wenet envriment:
Screenshot 2023-10-20 at 15 03 19

TODO: (in this pr)
- [ ] model.decode, 
- [ ]  支持cli
- [ ] 自动化导出（下载阿里模型）包含：词典，cmvn
- [ ] 精度校验

TODO： （future pr）
- [ ] fintune or train  from ali paraformer

NOTE： streaming paraformer not in current plan

### CODE REVIEW

**Comment by Mddct** [comment]
Screenshot 2023-10-24 at 18 37 25
cli work

**Comment by Mddct** [comment]
Screenshot 2023-10-30 at 17 26 47
recognize.py works

**Comment by Mddct** [comment]
decode info: batch_size=10
aishell:
| model  | greedy_search| beam_search|
|  ----  | ----  |----  |
| wenet-ali-paraformer |1.97 | 1.97|
| funasr-paraformer | | |

**Comment by robin1001** [comment]
assets 是必须的吗？是否可以放到模型中？

File: wenet/cli/transcribe.py [hunk 1]
```diff
@@ -40,14 +41,24 @@ def get_args():
     parser.add_argument('--align',
                         action='store_true',
                         help='force align the input audio and transcript')
-    parser.add_argument('--label', type=str, help='the input label to align')
+    parser.add_argument('--label',
+                        type=bool,
```
Comment (robin1001) [inline wenet/cli/transcribe.py:45]: label 是 str 类型，非 bool，这个修改是为了？

**Comment by Mddct** [comment]
> assets 是必须的吗？是否可以放到模型中？

导出模型的时候，需要用assets里边的文件，目前是我脚本转的
 可以等后边自动从funasr里边的conf力转成wenet，然后再删掉

**Review by robin1001 — APPROVED** [review]

**Comment by donstang** [comment]
streaming paraformer 有计划支持吗？

**Comment by Mddct** [comment]
> streaming paraformer 有计划支持吗？

暂时没计划， streaming的paraformer指标上差了一些 也不是理想中的流模型，

感兴趣的话 可以请关注wenetspeech2.0的进展

File: wenet/cli/paraformer_model.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        units_path = os.path.join(model_dir, 'units.txt')
+        self.model = torch.jit.load(model_path)
+        symbol_table = read_symbol_table(units_path)
+        self.char_dict = {v: k for k, v in symbol_table.items()}
+        self.eos = 2
+
+    def transcribe(self, audio_file: str, tokens_info: bool = False) -> dict:
+        waveform, sample_rate = torchaudio.load(audio_file, normalize=False)
+        waveform = waveform.to(torch.float)
+        feats = kaldi.fbank(waveform,
```
Comment (TeaPoly) [inline wenet/cli/paraformer_model.py:25]: The default window in the FunASR frontend is `hamming`. You can find more details [here](https://github.com/modelscope/FunASR/blob/42479a68cb3b2a95338d6c27a3ac9a5e66c60405/funasr/frontends/wav_frontend.py#L134). However, the default window in `kaldi.fbank` is `povey`, as specified [here](https://pytorch.org/audio/stable/generated/torchaudio.compliance.kaldi.fbank.html#torchaudio.compliance.kaldi.fbank). This different window maybe a little mismatch. As mentioned in line 44 of [this document](https://kaldi-asr.org/doc/feature-window_8h_source.html):
> "povey" is a window I made to be similar to Hamming but to go to zero at the edges, it's pow((0.5 - 0.5cos(n/N2*pi)), 0.85) I just don't think the Hamming window makes sense as a windowing function.
